### PR TITLE
docs: LHE file writing examples

### PR DESCRIPTION
**Pull request by APN-Pucky**
This adds two notebooks one showing how to write custom LHE files with a simple Monte Carlo event generator example and one filtering out events with a Z boson in the event record.

Further, the previously empty `class LHEFile` now contains two fields: `info` and `events` with a write function to a output stream. Contrary to the previous conversion to a string and then writing the string to the file, this approach allows us to write large filtered files directly per event to disk without consuming much memory.

One minor source for confusion might still remain: `write_lhe_file` takes a `init` and `events` while `read_lhe_file` returns a `LHEFile`. I'd change `write_lhe_file`, introduced in recent 0.9.0, to only require an LHEFile. The same then applies to `write_lhe_string`.

Closes: #251

### CODE REVIEW

**Comment by eduardo-rodrigues** [comment]
Hello @APN-Pucky. This  sounds great. I will try and have a proper look later in the week.

May I suggest that you also update the README with a paragraph or so about writing? One could even explicitly mention the existence of notebooks for how to get started with examples.

File: README.md [hunk 1]
```diff
@@ -59,6 +61,78 @@ event.graph.render(filename="test", format="png", cleanup=True)
 event.graph.render(filename="test", format="pdf", cleanup=True)
 ```
 
+### Writing
+
+For a full example see [write](examples/write_monte_carlo_example.ipynb) or [filter](examples/filter_events_example.ipynb).
+The values in the sketch below are intentionally left empty since they depend on the use-case.
+The data structure of pylhe is:
```
Comment (eduardo-rodrigues) [inline README.md:68]:
```suggestion
The data structure of `pylhe` is:
```

File: src/pylhe/__init__.py [hunk 2]
```diff
@@ -369,6 +369,42 @@ def fromstring(cls, string):
...
+        output_stream.write(
+            f"<LesHouchesEvents version=\"{self.init['LHEVersion']}\">\n"
+        )
+        output_stream.write(self.init.tolhe() + "\n")
+        for e in self.events:
+            output_stream.write(e.tolhe(rwgt=rwgt, weights=weights) + "\n")
+        output_stream.write("</LesHouchesEvents>")
+        return output_stream
+
+    def tolhe(self, rwgt=True, weights=False) -> str:
```
Comment (eduardo-rodrigues) [inline src/pylhe/__init__.py:390]: I'm a bit on the fence, here. Given the way we write methods I would be tempted to have this as `to_lhe`.

Comment (eduardo-rodrigues) [reply src/pylhe/__init__.py:389]: I see. OK, let's leave it, then.

**Comment by eduardo-rodrigues** [comment]
Hi @APN-Pucky, I reckon you can go ahead and then release a new version so that we have support for 3.13 out there :-). Advance thanks.

**Comment by APN-Pucky** [comment]
Good I'll try to follow https://www.youtube.com/watch?v=tsAWDumpcW8 and write a developer.md for #177 afterwards.

**Pull request merged**